src/main/java/edu/harvard/iq/dataverse/ManageFilePermissionsPage.java

```diff
@@ -390,6 +428,12 @@ private void grantAccessToRequests(AuthenticatedUser au, List<DataFile> files) {
         for (DataFile file : files) {
             if (assignRole(au, file, fileDownloaderRole)) {                
                 file.getFileAccessRequesters().remove(au);
+                List<FileAccessRequest> fileAccessRequests = fileAccessRequestService.findAllByAuthenticatedUserIdAndRequestState(au.getId(), FileAccessRequest.RequestState.CREATED);
```

Comment:
all access requests for this user? what if they've created requests for other datasets?

Reply:
Yes, this was a bug. I have fixed it.